nerfstudio/utils/tensor_dataclass.py

```diff
@@ -195,6 +195,28 @@ def ndim(self) -> int:
         """Returns the number of dimensions of the tensor dataclass."""
         return len(self._shape)
 
+    def cat(self: TensorDataclassT, other: TensorDataclassT, dim: int = 0) -> TensorDataclassT:
+        """Returns a new TensorDataclass with the other TensorDataclass's values concatenated along the specified dimension.
+
+        Args:
+            other: The other TensorDataclass to concatenate with.
+            dim: The dimension along which to concatenate the tensors.
+
+        Returns:
+            A new TensorDataclass with the tensors concatenated along the specified dimension.
+        """
+        if not isinstance(other, TensorDataclass):
+            raise TypeError("Expected 'other' to be a TensorDataclass.")
+
+        def tensor_fn(field_name, x):
+            return torch.cat([x, getattr(other, field_name)], dim=dim)
+
+        dataclass_fn = None
+
+        custom_tensor_dims_fn = None
```

Comment:
Could someone more familiar help me complete this? @tancik @jake-austin

I'm not too sure what the functions should look like.